src/store.ts

```diff
@@ -871,25 +871,46 @@ export default class IndexedFormula extends Formula { // IN future - allow pass
   }
 
   /**
-   * Removes all statements in a doc, along with the related metadata including request/response
+   * Removes all metadata
    * @param doc - The document / graph
    */
-  removeDocument(doc: Quad_Graph): IndexedFormula {
-    const meta = this.sym('chrome://TheCurrentSession') // or this.rdfFactory.namedNode('chrome://TheCurrentSession')
-    const linkNamespaceURI = 'http://www.w3.org/2007/ont/link#' // alain
-    // remove request/response and metadata
-    const requests = this.statementsMatching(undefined, this.sym(`${linkNamespaceURI}requestedURI`), this.rdfFactory.literal(doc.value), meta).map(st => st.subject)
+  removeMetadata(doc: Quad_Graph): IndexedFormula {
+    const meta = this.fetcher?.appNode // this.sym('chrome://TheCurrentSession')
+    const linkNamespaceURI = 'http://www.w3.org/2007/ont/link#'
+    const kb = this
+    // removeMatches() --> removeMany() --> remove() fails on Collection
+    function removeBySubject (subject) {
+      const sts = kb.statementsMatching(subject, null, null, meta)
+      // console.log(sts)
+      for (var i = 0; i < sts.length; i++) {
+        kb.removeStatement(sts[i])
+      }
+    }
+    const requests = this.statementsMatching(null, this.sym(`${linkNamespaceURI}requestedURI`), this.rdfFactory.literal(doc.value), meta).map(st => st.subject)
     for (var r = 0; r < requests.length; r++) {
       const request = requests[r]
-      if (request !== undefined) {
-        this.removeMatches(request, null, null, meta)
+      if (request != null) { // null or undefined
```

Comment:
The comment says "or undefined" but the code onyl checks for null. Why?

Reply:
Loose inequality does it
It is used in other places in rdflib

Reply:
**** I am not sure I understand this - if I create a file containing only `if(foo != null) console.log(1); ` I get `ReferenceError: foo is not **defined**`.  So I don't see how this can work.  I so much miss perl in this regard where you could just say if(foo) and that would cover null, undefined, and empty.